### ISSUE TITLE
FormHandler prematurely updates decimal NumberInput

### ISSUE THREAD

**Issue by t0rb3n** [mantinedev/mantine#3657]
### What package has an issue

@mantine/form

### Describe the bug

When using `NumberInput` in combination with `useForm` the values are updated too fast and resulting in not being able to enter any numbers. For integers this is fine but for decimal inputs this results in being able to write the first digit and then having to move the cursor to delete and write a new digit. Also if you keep trying to write a number > 5 at the back the number slowly grows.
I hope the bug is understandable.

Steps to reproduce: 
1. Go to [CodeSandbox](https://codesandbox.io/s/blissful-dust-w2djm6)
2. Start writing numbers in any field
3. Get instantly moved to back of the decimal digits resulting in not being able to write any more.

### What version of @mantine/hooks page do you have in package.json?

6.0.0

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/blissful-dust-w2djm6

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by bulicmatko** [comment]
Having the same problem with the `formatter` and `parser` props.
I think that a formatted number should not be stored as a value of the input.

Formatting should be applied only after blur (in a way that does not affect the input value).

When the user focuses on the input, formatting should be removed so the user is free to type in a number in a browser native way. After input is done and the user moves away from NumberInput, formatting can be applied without messing with the input value.

**Comment by rtivital** [comment]
Duplicate of #3663
